libipl/p10/ipl0.C

```diff
@@ -682,7 +683,14 @@ static int ipl_updatehwmodel(void)
 	    (!fs::exists(BOOTTIME_GUARD_INDICATOR)))
 		apply_fco_override();
 
-	process_guard_records();
+	try {
+		process_guard_records();
+	} catch (const openpower::guard::exception::GuardException &ex) {
+		// capture the exception to use it in the PEL
+		char *exceptionReason = const_cast<char *>(ex.what());
+		ipl_error_callback({IPL_ERR_GUARD_FILE_READ, exceptionReason});
+		throw ex;
```

Comment:
We should not throw the exception because we should continue the IPLing if any exception while processing the guard records.

Reply:
No. Actually we discussed this and felt it is better to put the host into quiesced state if we cant read the guard files. As there might be guard records that could point to unrecoverable failures and it may be advisable to continue with the IPLing

Reply:
Sorry, I am not following...
Do we need to continue the IPL or Do we need to stop it?

Reply:
sorry. Updated comment. We need to stop it